test/integration/gateways/sagepay_test.exs

```diff
@@ -0,0 +1,132 @@
+defmodule Gringotts.Integration.Gateways.SagePayTest do
+  # Integration tests for the Sagepay 
+
+  use ExUnit.Case, async: false
+  alias Gringotts.Gateways.SagePay
+  alias Gringotts.{CreditCard, Response}
+
+  @card %CreditCard{
+    number: "4484000000002",
+    month: 3,
+    year: 20,
+    first_name: "SAM",
+    last_name: "JONES",
+    verification_code: "123",
+    brand: "VISA"
+  }
+
+  @amount Money.new(100, :GBP)
+
+  @bad_card %CreditCard{
+    number: "4929000005559",
+    month: 3,
+    year: 20,
+    first_name: "SAM",
+    last_name: "JONES",
+    verification_code: "123",
+    brand: "VISA"
+  }
+
+  @address %{
+    address1: "407 St. John Street",
+    city: "London",
+    postalCode: "EC1V 4AB",
+    country: "GB"
+  }
+
+  @opts [
+    config: %{
+      auth_id:
+        "aEpZeHN3N0hMYmo0MGNCOHVkRVM4Q0RSRkxodUo4RzU0TzZyRHBVWHZFNmhZRHJyaWE6bzJpSFNyRnliWU1acG1XT1FNdWhzWFA1MlY0ZkJ0cHVTRHNocktEU1dzQlkxT2lONmh3ZDlLYjEyejRqNVVzNXU=",
+      vendor: "sandbox"
+    },
+    transactionType: "Deferred",
+    description: "Demo Payment",
+    customerFirstName: "Sam",
+    customerLastName: "Jones",
+    billingAddress: @address
+  ]
+
+  @bad_opts [
+    config: %{
+      auth_id:
+        "aEpZeHN3N0hMYmo0MGNCOHVkRVM4Q0RSRkxodUo4RzU0TzZyRHBVWHZFNmhZRHJyaWE6bzJpSFNyRnliWU1acG1XT1FNdWhzWFA1MlY0ZkJ0cHVTRHNocktEU1dzQlkxT2lONmh3ZDlLYjEyejRqNVVzNXU=",
+      vendor: "sandbox"
+    },
+    transactionType: "Deferred",
+    vendorTxCode: "demotransaction-20",
+    description: "Demo Payment",
+    customerFirstName: "Sam",
+    customerLastName: "Jones",
+    billingAddress: @address
+  ]
+
+  @payment_id "T6569400-1516-0A3F-E3FA-7F222CC79221"
+
+  @moduletag :integration
+
+  setup_all do
```

Comment:
This block is not required, the application config is fetched by `Gringotts` which is not used here.